### ISSUE TITLE
Create a Number class & Scalar class

### ISSUE THREAD

**Issue by schwern** [pdl/Test-Proto#33]
There is a String class but no Number class.  Given that Perl operators act differently on numbers vs strings, and that you want to do different things on numbers and strings, this seems like an opportunity to differentiate functionality.

There's also no Scalar class.  This would represent the sort of ambiguous thing that might be used as a number OR a string and it would have the methods of both.

Or maybe there just needs to be a Scalar class, because there's nothing that says you can't do string checks on a number...

**Comment by pdl** [comment]
Yeah, at the moment effectively the Scalar class is `Test::Proto::Base`, and the String class is `Test::Proto::String`. But I'm not happy about the division of labour.
